lib/sentry.service.ts

```diff
@@ -55,39 +54,49 @@ export class SentryService extends ConsoleLogger implements OnApplicationShutdow
   log(message: string, context?: string) {
     message = `${this.app} ${message}`;
     try {
+      if (this.opts && this.opts.logger) {
+        this.opts.logger.log(message, context);
+      }
       Sentry.captureMessage(message, Sentry.Severity.Log);
-      super.log(message, context);
     } catch (err) {}
   }
 
   error(message: string, trace?: string, context?: string) {
     message = `${this.app} ${message}`;
     try {
-      super.error(message, trace, context);
+      if (this.opts && this.opts.logger) {
+        this.opts.logger.error(message, trace, context);
+      }
       Sentry.captureMessage(message, Sentry.Severity.Error);
     } catch (err) {}
   }
 
   warn(message: string, context?: string) {
     message = `${this.app} ${message}`;
     try {
-      super.warn(message, context);
+      if (this.opts && this.opts.logger) {
+        this.opts.logger.warn(message, context);
+      }
       Sentry.captureMessage(message, Sentry.Severity.Warning);
     } catch (err) {}
   }
 
   debug(message: string, context?: string) {
     message = `${this.app} ${message}`;
     try {
-      super.debug(message, context);
+      if (this.opts && this.opts.logger && this.opts.logger.debug) {
+        this.opts.logger.debug(message, context);
+      }
       Sentry.captureMessage(message, Sentry.Severity.Debug);
     } catch (err) {}
   }
 
   verbose(message: string, context?: string) {
     message = `${this.app} ${message}`;
     try {
-      super.verbose(message, context);
+      if (this.opts && this.opts.logger && this.opts.logger.verbose) {
```

Comment:
Is there any change `verbose` is not defined on the logger? Isn't that part of the interface?

(same for debug above)

Reply:
As you can see the `NestJS`'s `LoggerService` interface declares `debug` and `verbose` methods as optional. `ConsoleLogger` implements all methods, but some custom loggers may omit these optional methods. This is just to be safe. Also, TS compiler suggested to check the availability of these methods.

```typescript
export interface LoggerService {
    /**
     * Write a 'log' level log.
     */
    log(message: any, ...optionalParams: any[]): any;
    /**
     * Write an 'error' level log.
     */
    error(message: any, ...optionalParams: any[]): any;
    /**
     * Write a 'warn' level log.
     */
    warn(message: any, ...optionalParams: any[]): any;
    /**
     * Write a 'debug' level log.
     */
    debug?(message: any, ...optionalParams: any[]): any;
    /**
     * Write a 'verbose' level log.
     */
    verbose?(message: any, ...optionalParams: any[]): any;
    /**
     * Set log levels.
     * @param levels log levels
     */
    setLogLevels?(levels: LogLevel[]): any;
}```

Reply:
Ah, I missed that, makes sense!